bundles/mapping/mapmodule/plugin/vectorlayer/VectorLayerPlugin.ol3.js

```diff
@@ -439,10 +411,14 @@ Oskari.clazz.define(
             var olLayer = me._olLayers[layer.getId()];
             if(!olLayer) {
                 olLayer = new olLayerVector({
-                    name: me._olLayerPrefix + layer.getId(),
-                    id: layer.getId(),
                     source: new olSourceVector()
                 });
+                // Set oskari properties
+                const silent = true;
+                olLayer.set(LAYER_ID, layer.getId(), silent);
+                olLayer.set(LAYER_TYPE, layer.getLayerType(), silent);
+                olLayer.set(LAYER_HOVER, layer.getHoverOptions(), silent);
+                olLayer.set(LAYER_NAME, me._olLayerPrefix + layer.getId(), silent);

```

Comment:
Moving the custom properties from `options` into layer object properties is actually even more problematic. Here we are for example setting the the properties "id" and "name" of the OL layer object. I think there is a high probability OL might use these properties for something in the future resulting in a conflict. 

This is a potential WTF for any future developers. The solution is not to attach these custom data to the OL layer object in any way.

Reply:
The newly added properties are prefixed with "oskari" to reduce propability of Openlayers ever using the same property names. Properties "id" and "name" are part of the old implementation and the task at hand is not to refactor existing implementation. Extra properties have been moved away from the constructor to clarify, that these are not native OL properties.

Reply:
[OL BaseObject.set/get](https://openlayers.org/en/latest/apidoc/module-ol_Object-BaseObject.html) is widely used in the codebase already.

Reply:
Jan has very good points for conflicting names but in my opinion it's ok to use the setters now that they have the oskari-prefixed names. Perhaps it's a bit overkill to have them as observables, but looks like OL docs agree with using the setter like this ("You can add your own observable properties with object.set('prop', 'value')") and it reduces complexity for having the layer-related props on the layer itself. 

We could of course get these from the OskariLayer as well where needed. Not sure if there would be some difference in performance on hover. Perhaps not if the OskariLayer-instance is stored on the hover handler and not fetched again and again for each feature from the service.